entity-framework/core/miscellaneous/testing/index.md

```diff
@@ -1,16 +1,106 @@
+
+Personally, I've never found it a problem having a database service running on my dev machine and I would generally recommend using Developer Edition instead.
+However, it may be appropriate for some people, especially on less powerful dev machines.  
+
+## Approach 2: SQLite
+
+EF Core tests the SQL Server provider primarily by running it against a local SQL Server instance.
+These tests run tens of thousands of queries in a couple of minutes on a fast machine.
+This illustrates that using the real database system can be a performant solution.
+It is a myth that using some lighter-weight database is the only way to run tests quickly.
+
+That being said, what if for whatever reason you can't run tests against something close to your production database system?
+The next best choice is to use something with similar functionality.
+This usually means another relational database, for which [SQLite](https://sqlite.org/index.html) is the obvious choice.
+
+SQLite is a good choice because:
+* It runs in-process with your application and so has low overhead.
+* It uses simple, automatically created files for databases, and so doesn't require database management.
+* It has an in-memory mode that avoids even the file creation.
+
+However, remember that:
+* SQLite inevitability doesn't support everything that your production database system does.
+* SQLite will behave differently than your production database system for some queries.
+
+So if you do use SQLite for some testing, make sure to also test against your real database system.
+
+See [Testing with SQLite](xref:core/miscellaneous/testing/sqlite) for EF Core specific guidance. 
+
+## Approach 3: The EF Core in-memory database
+
+EF Core comes with an in-memory database that we use for internal testing of EF Core itself.
+This database is in general **not suitable as a substitute for testing applications that use EF Core**. Specifically:
+* It is not a relational database
+* It doesn't support transactions
+* It is not optimized for performance
+
+None of this is very important when testing EF Core internals because we use it specifically where the database is irrelevant to the test.
+On the other hand, these things tend to be very important when testing an application that uses EF Core.
+
+## Unit testing
+
+Consider testing a piece of business logic that might need to use some data from a database, but is not inherently testing the database interactions.
+One option is to use a [test double](https://en.wikipedia.org/wiki/Test_double) such as a mock or fake.
+
+We use test doubles for internal testing of EF Core.
+However, we never try to mock DbContext or IQueryable.
+Doing so is difficult, cumbersome, and fragile.
+Don't do it.
+
+Instead we use the in-memory database when unit testing something that uses DbContext.
+In this case using the in-memory database is appropriate because the test is not dependent on database behavior.
+Just don't do this to test actual database queries or updates.   
```

Comment:
Ambiguous what "this" is referring to. In general this sentence is low value

Reply:
Will re-word or remove in next PR.